tests/unit/test_pipeline.py

```diff
@@ -29,60 +29,59 @@ def test_get_step_id(default_config, mocker):
 
 
-def test_get_input_files(default_config, mocker, test_dir):
+def test_get_input_files(default_config, mocker, test_dir, results_dir):
     mocker.patch("linker.implementation.Implementation.validate", return_value={})
     pipeline = Pipeline(default_config)
-    assert pipeline.get_input_files(pipeline.implementations[0], Path("/tmp")) == [
+    assert pipeline.get_input_files(pipeline.implementations[0]) == [
         test_dir + "/input_data1/file1.csv",
         test_dir + "/input_data2/file2.csv",
     ]
-    assert pipeline.get_input_files(pipeline.implementations[1], Path("/tmp")) == [
-        "/tmp/intermediate/1_step_1/result.parquet"
+    assert pipeline.get_input_files(pipeline.implementations[1]) == [
+        str(results_dir / "intermediate/1_step_1/result.parquet")
     ]
 
 
-def test_get_output_dir(default_config, mocker):
+def test_get_output_dir(default_config, mocker, results_dir):
     mocker.patch("linker.implementation.Implementation.validate", return_value={})
     pipeline = Pipeline(default_config)
-    assert pipeline.get_output_dir(pipeline.implementations[0], Path("/tmp")) == Path(
-        "/tmp/intermediate/1_step_1"
+    assert pipeline.get_output_dir(pipeline.implementations[0]) == Path(
+        results_dir / "intermediate/1_step_1"
     )
-    assert pipeline.get_output_dir(pipeline.implementations[1], Path("/tmp")) == Path("/tmp")
+    assert pipeline.get_output_dir(pipeline.implementations[1]) == results_dir
 
 
-def test_get_diagnostic_dir(default_config, mocker):
+def test_get_diagnostic_dir(default_config, mocker, results_dir):
     mocker.patch("linker.implementation.Implementation.validate", return_value={})
     pipeline = Pipeline(default_config)
-    assert pipeline.get_diagnostics_dir(pipeline.implementations[0], Path("/tmp")) == Path(
-        "/tmp/diagnostics/1_step_1"
+    assert pipeline.get_diagnostics_dir(pipeline.implementations[0]) == Path(
+        results_dir / "diagnostics/1_step_1"
     )
-    assert pipeline.get_diagnostics_dir(pipeline.implementations[1], Path("/tmp")) == Path(
-        "/tmp/diagnostics/2_step_2"
+    assert pipeline.get_diagnostics_dir(pipeline.implementations[1]) == Path(
+        results_dir / "diagnostics/2_step_2"
     )
 
 
 @pytest.mark.parametrize("computing_environment", ["local", "slurm"])
 def test_build_snakefile(default_config_params, mocker, test_dir, computing_environment):
+    results_dir = test_dir + "/results_dir"
     config_params = default_config_params
     if computing_environment == "slurm":
-        config_params.update(
-            {"computing_environment": Path(f"{test_dir}/spark_environment.yaml")}
-        )
+        config_params["computing_environment"] = Path(f"{test_dir}/spark_environment.yaml")
+
     config = Config(**config_params)
     mocker.patch("linker.implementation.Implementation.validate", return_value={})
     pipeline = Pipeline(config)
-    with TemporaryDirectory() as snake_dir:
```

Comment:
Not sure I understand what changed here. How are you getting around needing this context manager?

Reply:
Oh is it b/c test_dir is a tmpdir?